File: ios/LaneShadow/Views/Molecules/LSFilterChip.swift
import LaneShadowTheme
import SwiftUI

struct LSFilterChipResolvedStyle {
    let backgroundToken: String
    let borderToken: String
    let backgroundColor: Color
    let borderColor: Color
    let textColor: ContentColor
}

public struct LSFilterChip: View {
    public static let tapAccessibilityIdentifier = "lsfilterchip-toggle"

    @Environment(\.theme) private var theme

    let label: String
    let selected: Bool
    private let onToggle: () -> Void

    var resolvedStyle: LSFilterChipResolvedStyle {
        if selected {
            LSFilterChipResolvedStyle(
                backgroundToken: "color.signal.default",
                borderToken: "color.signal.default",
                backgroundColor: LaneShadowTheme.color.signal.default,
                borderColor: LaneShadowTheme.color.signal.default,
                textColor: .onSignal
            )
        } else {
            LSFilterChipResolvedStyle(
                backgroundToken: "color.surface.card",
                borderToken: "color.border.default",
                backgroundColor: LaneShadowTheme.color.surface.card,
                borderColor: LaneShadowTheme.color.border.default,
                textColor: .secondary
            )
        }
    }

    public init(
        label: String,
        selected: Bool,
        onToggle: @escaping () -> Void
    ) {
        self.label = label
        self.selected = selected
        self.onToggle = onToggle
    }

    public var body: some View {
        let style = resolvedStyle

        Button(action: { Self.dispatch(onToggle) }) {
            LSPill(size: .md) {
                LSText(label, variant: .label.md, color: style.textColor)
                    .padding(.horizontal, theme.space.xs)
                    .background(
                        RoundedRectangle(cornerRadius: theme.radius.full, style: .continuous)
                            .fill(style.backgroundColor)
                    )
                    .overlay(
                        RoundedRectangle(cornerRadius: theme.radius.full, style: .continuous)
                            .stroke(style.borderColor, lineWidth: theme.borderWidth.hairline)
                    )
            }
        }
        .buttonStyle(.plain)
        .accessibilityIdentifier(Self.tapAccessibilityIdentifier)
        .accessibilityAddTraits(.isButton)
    }

    static func dispatch(_ action: () -> Void) {
        action()
    }
}

File: android/app/src/main/java/com/laneshadow/ui/molecules/LSFilterChip.kt
package com.laneshadow.ui.molecules

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.role
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.semantics.selected
import androidx.compose.ui.unit.dp
import com.laneshadow.theme.LocalLaneShadowTheme
import com.laneshadow.ui.atoms.LSPill
import com.laneshadow.ui.atoms.LSText
import com.laneshadow.ui.atoms.PillSize

@Composable
fun LSFilterChip(
    label: String,
    selected: Boolean,
    onToggle: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
    size: PillSize = PillSize.Md,
) {
    val theme = LocalLaneShadowTheme.current
    val style = resolveFilterChipStyle(selected = selected)
    val shape = RoundedCornerShape(theme.radius.full)

    LSPill(
        size = size,
        modifier = modifier
            .background(style.backgroundColor, shape)
            .border(1.dp, style.borderColor, shape)
            .semantics {
                role = Role.Button
                this.selected = selected
                contentDescription = label
            }
            .clickable(enabled = enabled, onClick = onToggle),
    ) {
        LSText(
            text = label,
            variant = style.labelVariant,
            color = style.labelColor,
        )
    }
}
